### ISSUE TITLE
Improve error message when constructor is not correctly called

### ISSUE THREAD

**Issue by scabug** [scala/bug#7387]
In the following code example it is not immediately obvious what the error message means:

```scala
scala> class X(i: Int, j: Int) { def this(i: Int) = this(i, 0) }
defined class X

scala> class Y extends X
<console>:11: error: overloaded method constructor X with alternatives:
  (i: Int)X <and>
  (i: Int,j: Int)X
 cannot be applied to ()
       class Y extends X
               ^
```

First, the message should not point to the extends keyword but to the point where the constructor is called (at X). Secondly, while the error message is completely correct, it doesn't contain the information it should.

I like something like the following better:

```scala
error: multiple constructors for X available:
  (i: Int)X <and>
  (i: Int,j: Int)X
 you need to call one of them while extending X
       class Y extends X
                       ^
```

Because the empty parameter list is not part of the source code, I think it is useful not to mention it in the error message, for all other cases the following should be fine:

```scala
error: multiple constructors for X available:
  (i: Int)X <and>
  (i: Int,j: Int)X
 it is not possible to call one of them with (Int, Int, Int) while extending X
       class Y extends X(1,2,3)
                       ^
```

**Comment by som-snytt** [comment]
It's an overloaded method constructor, what's the problem?

We don't currently have the technology to determine whether the user wrote parentheses, but there is a box at EPFL taped shut with "Do not open until 23rd century!" with a request inside for someone to travel back in time and install the required tech in an alien obelisk so we can open it when we are ready. Unfortunately, the obelisk was constructed on a continental plate that is now under a mile of ice, so they're just waiting for that to melt first.
